TLCGen.PluggedInItems/TLCGen.Generators.CCOL/CodeGeneration/Sources/CCOLGeneratorGenerateOvC.cs

```diff
@@ -650,18 +650,26 @@ private string GenerateOvCRijTijdScenario(ControllerModel c)
                             var ll = fc.Detectoren.Where(x => x.Type == Models.Enumerations.DetectorTypeEnum.Lang).ToList();
                             if (ll.Count <= kl.Count)
                             {
-                                int i = 0;
-                                foreach (var d in kl)
+                                if (kl.Count==0)
                                 {
-                                    sb.Append($"{ts}OVRijTijdScenario(ovFC{fc.Naam}, {_dpf}{d.Naam}, ");
-                                    if (i < ll.Count)
-                                    {
-                                        sb.AppendLine($"{_dpf}{ll[i].Naam}, {_tpf}{_tbtovg}{fc.Naam});");
-                                        ++i;
-                                    }
-                                    else
+                                    sb.Append($"{ts}OVRijTijdScenario(ovFC{fc.Naam}, NG, NG, NG ");
+                                }
+                                else
+                                {
+                                    int i = 0;
+                                    foreach (var d in kl)
                                     {
-                                        sb.AppendLine("NG, NG);");
+                                        sb.Append($"{ts}OVRijTijdScenario(ovFC{fc.Naam}, {_dpf}{d.Naam}, ");
+                                        if (i < ll.Count)
+                                        {
+                                            sb.AppendLine($"{_dpf}{ll[i].Naam}, {_tpf}{_tbtovg}{fc.Naam});");
+                                            ++i;
+                                        }
+                                        else
+                                        {
+                                            sb.AppendLine("NG, NG);");
+                                        }
+                                        break;
```

Comment:
waarom een break? dan zal een tweede lus of lussenpaar buiten beschouwing worden gelaten.